game-core/src/main/java/games/strategy/engine/auto/update/EngineVersionCheck.java

```diff
@@ -46,4 +40,45 @@ static void checkForLatestEngineVersionOut() {
       log.log(Level.SEVERE, "Error while checking for engine updates", e);
     }
   }
+
+  private static boolean isEngineUpdateCheckRequired() {
+    return isEngineUpdateCheckRequired(
+        LocalDate.now(),
+        ClientSetting.TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY,
+        ClientSetting.TRIPLEA_LAST_CHECK_FOR_ENGINE_UPDATE,
+        ClientSetting::flush);
+  }
+
+  @VisibleForTesting
+  static boolean isEngineUpdateCheckRequired(
+      final LocalDate now,
+      final GameSetting firstRunSetting,
+      final GameSetting updateCheckDateSetting,
+      final Runnable flushSetting) {
+    // check at most once per 2 days (but still allow a 'first run message' for a new version of TripleA)
+    final boolean firstRun = firstRunSetting.booleanValue();
+    final String encodedUpdateCheckDate = updateCheckDateSetting.value();
+    if (!firstRun && !encodedUpdateCheckDate.trim().isEmpty()) {
+      final LocalDate updateCheckDate = parseUpdateCheckDate(encodedUpdateCheckDate);
+      if (updateCheckDate.until(now, ChronoUnit.DAYS) < 2) {
```

Comment:
Not sure about this one.
In order to be consistent It'd make sense to use `LocalDate#isBefore`/`LocalDate#isAfter` for this one too, however this would alter the behaviour.
However we could subtract a day from now to make `yesterday` and check using this day with the other method.